src/interpreter/index.ts

```diff
@@ -1614,14 +1615,24 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
 
     evaluate(this: Interpreter, expression: Expr.Expression): BrsType {
         this.location = expression.location;
+        this.stack.push(this.location);
         this.reportCoverageHit(expression);
-        return expression.accept<BrsType>(this);
+
+        let value = expression.accept<BrsType>(this);
+
+        this.stack.pop();
```

Comment:
Do we need to wrap this in a `try/finally` block?  I can't remember if `expression.accept` / `statement.execute` ever throw errors, but if they do it'll be really helpful to ensure the expression/statement that threw is removed from the stack.

Reply:
Great catch!